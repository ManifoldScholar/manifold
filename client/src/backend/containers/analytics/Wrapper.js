import React from "react";
import PropTypes from "prop-types";
import { useTranslation } from "react-i18next";
import { childRoutes } from "helpers/router";
import HeadContent from "global/components/HeadContent";

export default function AnalyticsWrapperContainer({ route }) {
  const { t } = useTranslation();

  return (
    <>
      <HeadContent
        title={`${t("titles.analytics")} | ${t("common.admin")}`}
        appendDefaultTitle
      />
      <main id="skip-to-main" tabIndex={-1}>
        <h1 className="screen-reader-text">{t("dashboard.title")}</h1>
        <section>
          <div className="container">{childRoutes(route)}</div>
        </section>
      </main>
    </>
  );
}

AnalyticsWrapperContainer.propTypes = {
  route: PropTypes.object
};
