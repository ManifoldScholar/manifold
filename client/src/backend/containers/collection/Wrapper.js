import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import connectAndFetch from "utils/connectAndFetch";
import Layout from "backend/components/layout";
import Dialog from "backend/components/dialog";
import Navigation from "backend/components/navigation";
import Utility from "global/components/utility";
import { entityStoreActions, notificationActions } from "actions";
import { select } from "utils/entityUtils";
import { collectionsAPI, requests } from "api";
import lh from "helpers/linkHandler";
import { childRoutes, RedirectToFirstMatch } from "helpers/router";
import navigation from "helpers/router/navigation";

import Authorize from "hoc/authorize";

const { request, flush } = entityStoreActions;

export class CollectionWrapperContainer extends PureComponent {
  static mapStateToProps = (state, ownPropsIgnored) => {
    return {
      collection: select(requests.beCollection, state.entityStore)
    };
  };

  static displayName = "Collection.Wrapper";

  static propTypes = {
    collection: PropTypes.object,
    dispatch: PropTypes.func,
    match: PropTypes.object,
    route: PropTypes.object,
    history: PropTypes.object
  };

  constructor(props) {
    super(props);
    this.state = {
      confirmation: null
    };
  }

  componentDidMount() {
    this.fetchCollection();
  }

  componentWillUnmount() {
    this.props.dispatch(flush(requests.beCollection));
  }

  fetchCollection = () => {
    const call = collectionsAPI.show(this.props.match.params.id);
    const collectionRequest = request(call, requests.beCollection);
    this.props.dispatch(collectionRequest);
  };

  closeDialog() {
    this.setState({ confirmation: null });
  }

  doPreview = event => {
    event.preventDefault();
    const previewUrl = lh.link(
      "frontendResourceCollection",
      this.props.collection.relationships.project.attributes.slug,
      this.props.collection.attributes.slug
    );
    const win = window.open(previewUrl, "_blank");
    win.focus();
  };

  doDestroy = () => {
    const call = collectionsAPI.destroy(this.props.collection.id);
    const options = { removes: this.props.collection };
    const collectionRequest = request(
      call,
      requests.beCollectionDestroy,
      options
    );
    this.props.dispatch(collectionRequest).promise.then(() => {
      this.notifyDestroy();
      this.redirectToProjectCollections();
    });
  };

  redirectToProjectCollections() {
    const projectId = this.props.collection.relationships.project.id;
    const redirectUrl = lh.link("backendProjectResourceCollections", projectId);
    this.props.history.push(redirectUrl);
  }

  notifyDestroy() {
    const notification = {
      level: 0,
      id: `COLLECTION_DESTROYED_${this.props.collection.id}`,
      heading: "The collection has been destroyed.",
      body: `${
        this.props.collection.attributes.title
      } has passed into the endless night.`,
      expiration: 5000
    };
    this.props.dispatch(notificationActions.addNotification(notification));
  }

  handleCollectionDestroy = event => {
    const heading = "Are you sure you want to delete this collection?";
    const message = "This action cannot be undone.";
    new Promise((resolve, reject) => {
      this.setState({
        confirmation: { resolve, reject, heading, message }
      });
    }).then(
      () => {
        this.doDestroy(event);
        this.closeDialog();
      },
      () => {
        this.closeDialog();
      }
    );
  };

  renderUtility() {
    return (
      <div>
        <button onClick={this.doPreview} className="button-bare-primary">
          <i className="manicon manicon-eye-outline" aria-hidden="true" />
          Preview{" "}
        </button>
        <button
          onClick={this.handleCollectionDestroy}
          className="button-bare-primary"
        >
          <i className="manicon manicon-trashcan" aria-hidden="true" />
          Delete
        </button>
      </div>
    );
  }

  renderRoutes() {
    const { collection } = this.props;
    return childRoutes(this.props.route, { childProps: { collection } });
  }

  render() {
    /* eslint-disable no-unused-vars */
    const { collection, match } = this.props;
    /* eslint-enable no-unused-vars */
    if (!collection) return null;
    const skipId = "skip-to-collection-panel";
    const secondaryLinks = navigation.collection(collection);

    return (
      <div>
        <Authorize
          entity={collection}
          failureFatalError={{
            body: "You are not allowed to edit this collection."
          }}
          ability="update"
        >
          <RedirectToFirstMatch
            from={lh.link("backendCollection", collection.id)}
            candidates={secondaryLinks}
          />
          {this.state.confirmation ? (
            <Dialog.Confirm {...this.state.confirmation} />
          ) : null}
          <Navigation.DetailHeader
            type="collection"
            backUrl={lh.link(
              "backendProjectResourceCollections",
              collection.relationships.project.id
            )}
            backLabel={
              collection.relationships.project.attributes.titlePlaintext
            }
            utility={this.renderUtility()}
            title={collection.attributes.title}
            secondaryLinks={secondaryLinks}
          />
          <Layout.BackendPanel
            sidebar={<Navigation.Secondary links={secondaryLinks} panel />}
          >
            <Utility.SkipLink skipId={skipId} />
            <div id={skipId}>{this.renderRoutes()}</div>
          </Layout.BackendPanel>
        </Authorize>
      </div>
    );
  }
}

export default connectAndFetch(CollectionWrapperContainer);