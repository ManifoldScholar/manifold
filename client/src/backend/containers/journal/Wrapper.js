import React, { useCallback } from "react";
import Layout from "backend/components/layout";
import withConfirmation from "hoc/withConfirmation";
import { journalsAPI } from "api";
import { childRoutes } from "helpers/router";
import lh from "helpers/linkHandler";
import navigation from "helpers/router/navigation";
import Authorize from "hoc/Authorize";
import {
  useFetch,
  useApiCallback,
  useNotification,
  useRedirectToFirstMatch
} from "hooks";
import { useTranslation } from "react-i18next";
import HeadContent from "global/components/HeadContent";
import PageHeader from "backend/components/layout/PageHeader";
import { RegisterBreadcrumbs } from "global/components/atomic/Breadcrumbs";

function JournalWrapper({ match, route, history, confirm, location }) {
  const { t } = useTranslation();
  const { data: journal, refresh } = useFetch({
    request: [journalsAPI.show, match.params.id]
  });
  const destroy = useApiCallback(journalsAPI.destroy, { removes: journal });

  const notifyDestroy = useNotification(j => ({
    level: 0,
    id: `JOURNAL_DESTROYED_${j.id}`,
    heading: t("notifications.journal_delete"),
    body: t("notifications.delete_entity_body", {
      title: j?.attributes?.title
    }),
    expiration: 5000
  }));

  const destroyAndRedirect = useCallback(() => {
    const redirect = () => history.push(lh.link("backendJournals"));
    destroy(journal.id).then(
      () => {
        notifyDestroy(journal);
        redirect();
      },
      () => redirect()
    );
  }, [destroy, history, journal, notifyDestroy]);

  const handleJournalDestroy = useCallback(() => {
    const heading = t("modals.delete_journal");
    const message = t("modals.confirm_body");
    confirm(heading, message, destroyAndRedirect);
  }, [destroyAndRedirect, confirm, t]);

  const utility = [
    {
      label: "actions.view",
      route: "frontendJournalDetail",
      slug: journal?.attributes.slug,
      icon: "eyeOpen32"
    },
    {
      label: "actions.delete",
      authorize: "delete",
      icon: "delete32",
      onClick: handleJournalDestroy
    }
  ];

  const renderRoutes = () => {
    return childRoutes(route, {
      childProps: { refresh, journal }
    });
  };

  useRedirectToFirstMatch({
    route: "backendJournal",
    id: journal?.id,
    slug: journal?.attributes.slug,
    candidates: journal ? navigation.journal(journal) : []
  });

  if (!journal) return null;

  const subpage = location.pathname.split("/")[4]?.replace("-", "_");

  const breadcrumbs = [
    {
      to: lh.link("backendJournals"),
      label: t("glossary.journal_title_case_other")
    },
    {
      to: lh.link("backendJournals", journal.id),
      label: journal.attributes.titlePlaintext
    }
  ];

  return (
    <div>
      <Authorize
        entity={journal}
        failureFatalError={{
          detail: t("journals.unauthorized_edit")
        }}
        ability={["read"]}
      >
        {subpage && (
          <HeadContent
            title={`${t(`titles.${subpage}`)} | ${
              journal.attributes.titlePlaintext
            } | ${t("common.admin")}`}
            appendDefaultTitle
          />
        )}
        <RegisterBreadcrumbs breadcrumbs={breadcrumbs ?? []} />
        <PageHeader
          type="journal"
          title={journal.attributes.titleFormatted}
          subtitle={journal.attributes.subtitle}
          actions={utility}
          secondaryLinks={navigation.journal(journal)}
          issues={journal.attributes.issuesNav}
        />
        <Layout.BackendPanel
          sidebar={
            <Layout.SecondaryNav
              links={navigation.journal(journal)}
              panel
              ariaLabel={t("journals.settings")}
            />
          }
        >
          <div>{renderRoutes()}</div>
        </Layout.BackendPanel>
      </Authorize>
    </div>
  );
}

export default withConfirmation(JournalWrapper);
