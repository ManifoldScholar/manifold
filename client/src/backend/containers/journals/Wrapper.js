import React from "react";
import PropTypes from "prop-types";
import { useTranslation } from "react-i18next";
import { childRoutes } from "helpers/router";
import Authorize from "hoc/Authorize";

export default function JournalsWrapper({ route }) {
  const { t } = useTranslation();
  return (
    <Authorize
      ability="update"
      entity={["journal"]}
      failureFatalError={{
        body: t("journals.unauthorized")
      }}
    >
      <div>
        <main id="skip-to-main" tabIndex={-1} className="backend-detail">
          {childRoutes(route)}
        </main>
      </div>
    </Authorize>
  );
}

JournalsWrapper.propTypes = {
  route: PropTypes.object
};
