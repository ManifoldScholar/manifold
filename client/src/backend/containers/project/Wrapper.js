import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import { withTranslation } from "react-i18next";
import connectAndFetch from "utils/connectAndFetch";
import Layout from "backend/components/layout";
import withConfirmation from "hoc/withConfirmation";
import { entityStoreActions } from "actions";
import { select } from "utils/entityUtils";
import { projectsAPI, requests } from "api";
import { childRoutes, RedirectToFirstMatch } from "helpers/router";
import lh from "helpers/linkHandler";
import navigation from "helpers/router/navigation";
import Authorize from "hoc/Authorize";
import get from "lodash/get";
import HeadContent from "global/components/HeadContent";
import { RegisterBreadcrumbs } from "global/components/atomic/Breadcrumbs";
import { getBreadcrumbs } from "./breadcrumbs";
import PageHeader from "backend/components/layout/PageHeader";

const { request, flush } = entityStoreActions;

export class ProjectWrapperContainer extends PureComponent {
  static mapStateToProps = state => {
    return {
      projectResponse: get(state.entityStore.responses, requests.beProject),
      project: select(requests.beProject, state.entityStore)
    };
  };

  static displayName = "Project.Wrapper";

  static propTypes = {
    projectResponse: PropTypes.object,
    project: PropTypes.object,
    dispatch: PropTypes.func,
    match: PropTypes.object,
    history: PropTypes.object,
    confirm: PropTypes.func.isRequired,
    route: PropTypes.object,
    location: PropTypes.object,
    t: PropTypes.func
  };

  static defaultProps = {
    confirm: (heading, message, callback) => callback()
  };

  componentDidMount() {
    this.fetchProject();
  }

  componentDidUpdate() {
    const {
      params: { id: nextId }
    } = this.props.match ?? {};
    const prevId = this.props.project?.id;
    if (nextId && prevId && nextId !== prevId) this.fetchProject();
  }

  componentWillUnmount() {
    this.props.dispatch(flush(requests.beProject));
  }

  fetchProject = () => {
    const call = projectsAPI.show(this.props.match.params.id);
    const options = { force: true };
    const projectRequest = request(call, requests.beProject, options);
    this.props.dispatch(projectRequest);
  };

  doDestroy = () => {
    const call = projectsAPI.destroy(this.props.project.id);
    const options = { removes: this.props.project };
    const projectRequest = request(call, requests.beProjectDestroy, options);
    this.props.dispatch(projectRequest).promise.then(() => {
      this.redirectToDashboard();
    });
  };

  redirectToDashboard() {
    this.props.history.push(lh.link("backend"));
  }

  handleProjectDestroy = () => {
    const { t } = this.props;
    const heading = t("modals.delete_project");
    const message = t("modals.confirm_body");
    this.props.confirm(heading, message, this.doDestroy);
  };

  get utility() {
    return [
      {
        label: "actions.view",
        route: "frontendProjectDetail",
        slug: this.props.project.attributes.slug,
        icon: "eyeOpen32"
      },
      {
        label: "actions.delete",
        authorize: "delete",
        icon: "delete32",
        onClick: this.handleProjectDestroy
      }
    ];
  }

  renderRoutes() {
    const { project, projectResponse } = this.props;
    const refresh = this.fetchProject;
    const updateProject = projectsAPI.update;
    return childRoutes(this.props.route, {
      childProps: { refresh, updateProject, project, projectResponse }
    });
  }

  render() {
    if (!this.props.project) return null;
    const { project, t } = this.props;
    const secondaryLinks = navigation.project(project);
    const isJournalIssue = project.attributes.isJournalIssue;

    const breadcrumbs = getBreadcrumbs(
      project,
      this.props.location.state,
      isJournalIssue,
      t
    );

    const subpage = location.pathname.split("/")[4]?.replace("-", "_");

    const parentProps = isJournalIssue
      ? {
          parentTitle: project.relationships.journal.attributes.titleFormatted,
          parentSubtitle: project.relationships.journal.attributes.subtitle,
          parentId: project.relationships.journal.id,
          issues: project.attributes.journalIssuesNav
        }
      : {};

    return (
      <Authorize
        entity={project}
        failureFatalError={{
          detail: t("projects.unauthorized_edit")
        }}
        ability={["update", "manageResources"]}
      >
        {subpage && (
          <HeadContent
            title={`${t(`titles.${subpage}`)} | ${
              project.attributes.titlePlaintext
            } | ${t("common.admin")}`}
            appendDefaultTitle
          />
        )}
        <RedirectToFirstMatch
          route={"backendProject"}
          id={project.id}
          slug={project.attributes.slug}
          candidates={secondaryLinks}
          state={this.props.location.state}
        />
        <RegisterBreadcrumbs breadcrumbs={breadcrumbs ?? []} />
        <PageHeader
          type={isJournalIssue ? "issue" : "project"}
          title={project.attributes.titleFormatted}
          subtitle={project.attributes.subtitle}
          texts={project.attributes.textsNav}
          actions={this.utility}
          secondaryLinks={secondaryLinks}
          {...parentProps}
        />
        <Layout.BackendPanel
          sidebar={
            <Layout.SecondaryNav
              links={secondaryLinks}
              panel
              ariaLabel={t("projects.settings")}
            />
          }
        >
          <div>{this.renderRoutes()}</div>
        </Layout.BackendPanel>
      </Authorize>
    );
  }
}

export default withTranslation()(
  withConfirmation(connectAndFetch(ProjectWrapperContainer))
);
