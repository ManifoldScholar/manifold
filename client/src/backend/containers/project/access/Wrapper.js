import React, { Component } from "react";
import PropTypes from "prop-types";
import { withTranslation } from "react-i18next";
import { childRoutes } from "helpers/router";
import lh from "helpers/linkHandler";
import PermissionsContainer from "backend/containers/permission";
import EntitlementsContainer from "backend/containers/entitlements";
import Authorize from "hoc/Authorize";
import Authorization from "helpers/authorization";
import Layout from "backend/components/layout";
import FormContainer from "global/containers/form";
import Form from "global/components/form";
import Hero from "backend/components/hero";
import withSettings from "hoc/withSettings";
import connectAndFetch from "utils/connectAndFetch";

class ProjectAccessWrapper extends Component {
  static mapStateToProps = state => {
    return {
      authentication: state.authentication
    };
  };

  static displayName = "Project.Access.Wrapper";

  static propTypes = {
    project: PropTypes.object,
    route: PropTypes.object,
    history: PropTypes.object,
    match: PropTypes.object,
    authentication: PropTypes.object,
    t: PropTypes.func
  };

  get defaultIsRestricted() {
    return this.props.settings.attributes.general.restrictedAccess === true;
  }

  get defaultIsOpen() {
    return !this.defaultIsRestricted;
  }

  render() {
    const { project, updateProject, t, authentication } = this.props;
    if (!project) return null;

    const authorization = new Authorization();

    const canGrantPermissions = authorization.authorizeAbility({
      authentication,
      entity: project,
      ability: "managePermissions"
    });

    const closeUrl = lh.link("backendProjectAccess", project.id);
    return (
      <>
        <Authorize
          entity={project}
          ability="update"
          failureNotification
          failureRedirect={lh.link("backendProject", project.id)}
        >
          {canGrantPermissions && (
            <Layout.BackendPanel flush>
              <PermissionsContainer.List entity={project} />
            </Layout.BackendPanel>
          )}
          <Layout.BackendPanel flush={!canGrantPermissions}>
            <EntitlementsContainer.List
              entity={project}
              preList={
                <div style={{ marginBottom: 44, marginTop: 22 }}>
                  <Hero.Block
                    title={t("projects.forms.access.title")}
                    description={t("projects.forms.access.description")}
                    initialVisible={this.defaultIsRestricted}
                  >
                    <FormContainer.Form
                      style={{ paddingTop: 24, paddingBottom: 24 }}
                      model={project}
                      name="backend-project-update"
                      update={updateProject}
                      className="form-secondary"
                    >
                      {this.defaultIsOpen && (
                        <Form.Switch
                          label={t("projects.forms.access.restricted_label")}
                          name="attributes[restrictedAccess]"
                          wide
                        />
                      )}
                      {this.defaultIsRestricted && (
                        <Form.Switch
                          label={t("projects.forms.access.open_access_label")}
                          name="attributes[openAccess]"
                          wide
                        />
                      )}
                      <Form.TextInput
                        label={t("projects.forms.access.notice_header_label")}
                        name="attributes[restrictedAccessHeading]"
                        placeholder={t(
                          "projects.forms.access.notice_header_placeholder"
                        )}
                        wide
                      />
                      <Form.TextArea
                        label={t("projects.forms.access.notice_body_label")}
                        name="attributes[restrictedAccessBody]"
                        placeholder={t(
                          "projects.forms.access.notice_body_placeholder"
                        )}
                      />
                      <Form.Save text={t("projects.forms.access.save")} />
                    </FormContainer.Form>
                  </Hero.Block>
                </div>
              }
            />
          </Layout.BackendPanel>
        </Authorize>
        {childRoutes(this.props.route, {
          drawer: true,
          drawerProps: { closeUrl, lockScroll: "always" },
          childProps: {
            entity: project,
            closeUrl,
            history: this.props.history
          }
        })}
      </>
    );
  }
}

export default withTranslation()(
  withSettings(connectAndFetch(ProjectAccessWrapper))
);
