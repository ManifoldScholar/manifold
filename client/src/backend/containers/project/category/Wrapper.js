import React, { Component } from "react";
import PropTypes from "prop-types";
import { withRouter } from "react-router-dom";
import { childRoutes } from "helpers/router";

class ProjectCategoryWrapperContainer extends Component {
  static displayName = "Project.Category.Wrapper";

  static propTypes = {
    project: PropTypes.object,
    history: PropTypes.object,
    route: PropTypes.object
  };

  render() {
    return (
      <div>{childRoutes(this.props.route, { childProps: this.props })}</div>
    );
  }
}

export default withRouter(ProjectCategoryWrapperContainer);
