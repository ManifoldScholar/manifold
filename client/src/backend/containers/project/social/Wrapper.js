import React, { Component } from "react";
import PropTypes from "prop-types";
import TwitterQueries from "./TwitterQueries";
import { childRoutes } from "helpers/router";
import lh from "helpers/linkHandler";

import Authorize from "hoc/Authorize";

export default class ProjectSocialWrapperContainer extends Component {
  static displayName = "Project.Social.Wrapper";

  static propTypes = {
    project: PropTypes.object,
    route: PropTypes.object,
    history: PropTypes.object,
    match: PropTypes.object
  };

  render() {
    const project = this.props.project;
    const closeUrl = lh.link("backendProjectSocial", project.id);

    return (
      <Authorize
        entity={project}
        ability="manageSocials"
        failureNotification
        failureRedirect={lh.link("backendProject", project.id)}
      >
        <section>
          {childRoutes(this.props.route, {
            drawer: true,
            drawerProps: { closeUrl }
          })}
          <TwitterQueries project={project} match={this.props.match} />
        </section>
      </Authorize>
    );
  }
}
