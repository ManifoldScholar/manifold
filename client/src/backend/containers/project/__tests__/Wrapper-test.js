import { ProjectWrapperContainer } from "../Wrapper";
import { project, route } from "./__fixtures__";
import { BreadcrumbsProvider } from "global/components/atomic/Breadcrumbs";

describe("backend/containers/project/Wrapper", () => {
  beforeEach(() => {
    testHelpers.startSession($dispatch, $user);
  });

  def("abilities", () => ({ update: true }));
  def("user", () => factory("user"));
  def("project", () => project($abilities));
  def("route", () => route());
  def("match", () => ({
    params: {}
  }));
  def("root", () => (
    <BreadcrumbsProvider>
      <ProjectWrapperContainer
        project={$project}
        dispatch={$dispatch}
        location={{ pathname: "/projects/1" }}
        route={$route}
        match={$match}
        t={key => key}
      />
    </BreadcrumbsProvider>
  ));

  it("matches the snapshot when rendered", () => {
    expect(render($withApp($root)).html()).toMatchSnapshot();
  });

  it("does not render a null value", () => {
    expect(render($withApp($root)).html()).not.toBeNull();
  });
});
