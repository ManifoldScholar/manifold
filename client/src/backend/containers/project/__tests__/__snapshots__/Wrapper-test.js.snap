// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`backend/containers/project/Wrapper matches the snapshot when rendered 1`] = `"<div class=\\"css-1k4if3i-Content ecdc3151\\"><div class=\\"css-1jayvsg-Row e12ljl8g17\\"><figure class=\\"css-kahbcu-Figure e12ljl8g16\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 48 48\\" width=\\"100%\\" height=\\"100%\\" class=\\"project-thumb-placeholder css-9v4qk6-ProjectIcon e12ljl8g6 project-thumb-placeholder--all project-thumb-placeholder--outlined\\" aria-hidden=\\"false\\"><g fill=\\"none\\" fill-rule=\\"evenodd\\"><polygon class=\\"project-thumb-placeholder__tile\\" fill=\\"#CBF7E6\\" points=\\"0 42 42 42 42 0 0 0\\"></polygon><g class=\\"project-thumb-placeholder__illustration\\" transform=\\"translate(9 8)\\"><g transform=\\"translate(20.683 2.244)\\"><polygon points=\\".161 2.954 .161 21.013 1.612 23.598 3.06 21.013 3.06 2.954\\"></polygon><path d=\\"M3.06081501,21.012797 L0.161633021,21.012797\\"></path><path stroke-linejoin=\\"round\\" d=\\"M0.162011257,2.95560225 L3.06119325,2.95560225 L3.06119325,1.53299812 C3.06119325,0.980713374 2.613478,0.532998124 2.06119325,0.532998124 L1.16201126,0.532998124 C0.609726507,0.532998124 0.162011257,0.980713374 0.162011257,1.53299812 L0.162011257,2.95560225 Z\\"></path></g><g transform=\\"translate(.195 11.463)\\"><polygon points=\\"14.898 .419 .985 .419 .985 11.282 8.141 11.282 8.141 14.378 10.844 11.282 14.898 11.282\\"></polygon><path d=\\"M12.5237854 3.46905816L3.32696735 3.46905816M12.5237854 8.00165403L3.32696735 8.00165403M12.5237854 5.73507242L3.32696735 5.73507242\\"></path></g><g transform=\\"translate(.195 .195)\\"><path d=\\"M11.8853223 5.12585966C11.8853223 6.3608015 10.8848871 7.36123677 9.64994522 7.36123677 8.41689456 7.36123677 7.41645929 6.3608015 7.41645929 5.12585966 7.41645929 3.89091782 8.41689456 2.89048255 9.64994522 2.89048255 10.8848871 2.89048255 11.8853223 3.89091782 11.8853223 5.12585966zM11.8853223 5.23573734L15.0417051 2.0793546C15.6109508 1.51010882 16.5849096 1.91293058 16.5849096 2.71857411M3.99814784 3.09718874L6.63067317.464663415C7.20181013-.106473546 8.17576886.2982394 8.17576886 1.10388293\\"></path><path d=\\"M5.22155347 5.12585966C5.22155347 6.3608015 4.2211182 7.36123677 2.98617636 7.36123677 1.7531257 7.36123677.752690432 6.3608015.752690432 5.12585966.752690432 3.89091782 1.7531257 2.89048255 2.98617636 2.89048255 4.2211182 2.89048255 5.22155347 3.89091782 5.22155347 5.12585966zM5.22155347 5.34334559L5.71136961 5.13720675C6.09906191 4.97267392 6.53781614 4.97267392 6.92739962 5.13720675L7.41532458 5.34334559\\"></path></g></g></g></svg></figure><div class=\\"css-afmy3i-TitleWrapper e12ljl8g4\\"><h1 class=\\"css-ytpeqb-Title e12ljl8g3\\">Rowan Test</h1><span class=\\"css-xbanu3-Subtitle e12ljl8g2\\">World's Greatest Dog</span></div></div><div class=\\"css-hdwovy-Row-DropdownRow e1yvm5522\\"><div class=\\"css-14hqmci-Positioner e2613d75\\"><button tabindex=\\"0\\" type=\\"button\\" aria-expanded=\\"false\\" aria-controls=\\"abcdef\\" class=\\"css-smivd9-Button e2613d77\\"><span>actions.actions</span><svg xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"manicon-svg css-klehzp-DropdownIcon e2613d76 svg-icon--disclosureDown24\\" width=\\"22\\" height=\\"22\\" viewBox=\\"0 0 24 24\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><polygon points=\\"20.374 9 21 9.78 12 17 3 9.78 3.626 9 12 15.718\\"></polygon></svg></button><div id=\\"abcdef\\" class=\\"css-1cn06ep-Wrapper e2613d74\\"><ul class=\\"css-5ovq0u-List e2613d73\\"><li><a class=\\"css-1tupqjh-Link e2613d71\\" href=\\"/projects/slug-AAAAAAAA-BBBB-CCCC-DDDD-EEEEEEEEEEEE\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"manicon-svg css-1juodu2-LinkIcon e2613d70 svg-icon--eyeOpen32\\" width=\\"20\\" height=\\"20\\" viewBox=\\"0 0 32 32\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path fill-rule=\\"evenodd\\" d=\\"M16.0027,12.509 C14.3487,12.509 13.0027,13.855 13.0027,15.509 C13.0027,17.163 14.3487,18.509 16.0027,18.509 C17.6567,18.509 19.0027,17.163 19.0027,15.509 C19.0027,13.855 17.6567,12.509 16.0027,12.509 M16.0027,19.509 C13.7967,19.509 12.0027,17.715 12.0027,15.509 C12.0027,13.303 13.7967,11.509 16.0027,11.509 C18.2087,11.509 20.0027,13.303 20.0027,15.509 C20.0027,17.715 18.2087,19.509 16.0027,19.509 M5.242,15.508 C6.262,16.807 10.683,22.015 16.003,22.015 C21.322,22.015 25.743,16.808 26.763,15.508 C25.743,14.208 21.322,9 16.003,9 C10.675,9 6.261,14.208 5.242,15.508 M16.003,23.015 C9.437,23.015 4.418,16.093 4.207,15.799 L4,15.508 L4.207,15.217 C4.418,14.923 9.437,8 16.003,8 C22.568,8 27.588,14.923 27.798,15.217 L28.006,15.508 L27.798,15.799 C27.588,16.093 22.568,23.015 16.003,23.015\\"></path></svg><span class=\\"css-1avywkn-LinkText e2613d72\\">actions.view</span></a></li></ul></div></div></div><div class=\\"css-18isdm2-UtilityRow e1yvm5521\\"><div class=\\"utility-button-group utility-button-group--inline\\"><a class=\\"utility-button\\" href=\\"/projects/slug-AAAAAAAA-BBBB-CCCC-DDDD-EEEEEEEEEEEE\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"manicon-svg utility-button__icon svg-icon--eyeOpen32\\" width=\\"26\\" height=\\"26\\" viewBox=\\"0 0 32 32\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path fill-rule=\\"evenodd\\" d=\\"M16.0027,12.509 C14.3487,12.509 13.0027,13.855 13.0027,15.509 C13.0027,17.163 14.3487,18.509 16.0027,18.509 C17.6567,18.509 19.0027,17.163 19.0027,15.509 C19.0027,13.855 17.6567,12.509 16.0027,12.509 M16.0027,19.509 C13.7967,19.509 12.0027,17.715 12.0027,15.509 C12.0027,13.303 13.7967,11.509 16.0027,11.509 C18.2087,11.509 20.0027,13.303 20.0027,15.509 C20.0027,17.715 18.2087,19.509 16.0027,19.509 M5.242,15.508 C6.262,16.807 10.683,22.015 16.003,22.015 C21.322,22.015 25.743,16.808 26.763,15.508 C25.743,14.208 21.322,9 16.003,9 C10.675,9 6.261,14.208 5.242,15.508 M16.003,23.015 C9.437,23.015 4.418,16.093 4.207,15.799 L4,15.508 L4.207,15.217 C4.418,14.923 9.437,8 16.003,8 C22.568,8 27.588,14.923 27.798,15.217 L28.006,15.508 L27.798,15.799 C27.588,16.093 22.568,23.015 16.003,23.015\\"></path></svg><span class=\\"utility-button__text\\">actions.view</span></a></div></div><nav class=\\"ecdc3150 css-8n39ik-Nav-SecondaryNav e1vnhept5\\"><button class=\\"css-iszmcy-Disclosure e1vnhept3\\"><div class=\\"css-1wmeakp-Active e1vnhept4\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"manicon-svg css-9dlcye-Icon e1vnhept2 svg-icon--disclosureDown16\\" width=\\"16\\" height=\\"16\\" viewBox=\\"0 0 16 16\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><polygon points=\\"8 9.702 13.356 5 13.999 5.733 8 10.999 2 5.733 2.643 5\\"></polygon></svg></div></button><ul class=\\"css-62htyi-List e1vnhept1\\"><li><a class=\\"css-slu7jg-Link e1vnhept0\\" href=\\"/backend/projects/AAAAAAAA-BBBB-CCCC-DDDD-EEEEEEEEEEEE/analytics\\">titles.analytics</a></li><li><a class=\\"css-slu7jg-Link e1vnhept0\\" href=\\"/backend/projects/AAAAAAAA-BBBB-CCCC-DDDD-EEEEEEEEEEEE/properties\\">titles.properties</a></li><li><a class=\\"css-slu7jg-Link e1vnhept0\\" href=\\"/backend/projects/AAAAAAAA-BBBB-CCCC-DDDD-EEEEEEEEEEEE/layout\\">titles.layout</a></li><li><a class=\\"css-slu7jg-Link e1vnhept0\\" href=\\"/backend/projects/AAAAAAAA-BBBB-CCCC-DDDD-EEEEEEEEEEEE/access\\">titles.access</a></li><li><a class=\\"css-slu7jg-Link e1vnhept0\\" href=\\"/backend/projects/AAAAAAAA-BBBB-CCCC-DDDD-EEEEEEEEEEEE/metadata\\">titles.metadata</a></li></ul></nav></div>"`;
